Automattic/wp-calypso
# Update wp-env and some ETK PHP fixes

As I began working @tyxla's comment here (https://github.com/Automattic/wp-calypso/pull/79097#issuecomment-1629648852), I realized it'd be good to update various wp-env/ETK things (which I've been putting off for a while).

## Proposed Changes
It's been a while since I touched wp-env in Calypso. This updates our config to make it compatible with new wp-env versions, which in turn means we're updating phpunit and setting a newer PHP version. As a result, there were a few other things to fix too. 

Importantly, there are some features in ETK which were not written to work in local environments. In fact, they crash when not running on Automattic infrastructure. I decided to leave fixes for these out of the PR to make it less risky, but the result is that some pages of wp-env do not work. I will follow up with teams to try to get this resolved.

## Local Testing Instructions
1. Run docker
2. `cd apps/editing-toolkit`
3. Build the plugin (`yarn build`)
6. After running `composer install`, run `yarn test:php` locally.

## Wpcom testing instructions
1. On your sandbox, run `install-plugin.sh etk update-wp-env`.
2. Sandbox a test site and load the "new page" editor
3. Verify the "page pattern picker" loads, and then use the three-dot info dropdown in the editor to check that the ETK version is the same as the one installed by install-plugin in step 1.
4. Verify that the help center and whats-new still load on wpcom.
